realm/realm-library/src/androidTest/java/io/realm/ManagedRealmCollectionTests.java

```diff
@@ -491,12 +490,14 @@ public void avg_partialNullRows() {
 
     @Test
     public void maxDate() {
-        assertEquals(new Date(YEAR_MILLIS * 20 * 4), collection.maxDate(AllJavaTypes.FIELD_DATE));
+        assertEquals(TEST_SIZE, collection.size());
+        assertEquals(new Date(YEAR_MILLIS * 20 * (TEST_SIZE/2 - 1)), collection.maxDate(AllJavaTypes.FIELD_DATE));
```

Comment:
It didn't but `4` is assuming a value of `TEST_SIZE` and the test would fail if `TEST_SIZE` was changed. I just made it a bit for robust.